Cannot load zip archived ROMs from the command line

## Description

Loading zip archived games works when loading them through the RetroArch UI. However, the same games will fail to load when running from the command line like so:
```
retroarch --libretro /usr/lib/libretro/snes9x_libretro.so krusty.zip --verbose
[INFO] RetroArch 1.9.5 (Git 6738fa57aa)
[INFO] === Build =======================================
[INFO] CPU Model Name: Intel(R) Core(TM) i7-8650U CPU @ 1.90GHz
[INFO] Capabilities:  MMX MMXEXT SSE SSE2 SSE3 SSSE3 SSE4 SSE4.2 AES AVX AVX2
[INFO] Built: Jun 14 2021
[INFO] Version: 1.9.5
[INFO] Git: 6738fa57aa
[INFO] =================================================
[INFO] [Input]: Found input driver: "x".
[INFO] [Core]: Loading dynamic libretro core from: "/usr/lib/libretro/snes9x_libretro.so"
[INFO] [Overrides]: No core-specific overrides found at "~/.config/retroarch/config/Snes9x/Snes9x.cfg".
[INFO] [Overrides]: No content-dir-specific overrides found at "~/.config/retroarch/config/Snes9x/.cfg".
[INFO] [Overrides]: No game-specific overrides found at "~/.config/retroarch/config/Snes9x/krusty.cfg".
[INFO] [Environ]: GET_CORE_OPTIONS_VERSION.
[INFO] [Environ]: GET_LANGUAGE: "0".
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_INTL.
[INFO] [Environ]: SET_CONTROLLER_INFO.
[INFO] [Remaps]: Remap directory: "~/.config/retroarch/config/remaps".
[INFO] [Overrides]: Redirecting save file to "~/.config/retroarch/saves/krusty.srm".
[INFO] [Overrides]: Redirecting save state to "~/.config/retroarch/states/krusty.state".
[INFO] [Environ]: GET_LOG_INTERFACE.
[INFO] [Environ]: SYSTEM_DIRECTORY: "~/.config/retroarch/system".
[INFO] [Environ]: GET_SAVE_DIRECTORY.
[INFO] [Environ]: SET_SUPPORT_ACHIEVEMENTS: yes.
[INFO] [Environ]: PERFORMANCE_LEVEL: 12.
[INFO] [CONTENT LOAD]: Loading content file: krusty.zip#Krusty's Super Fun House (U) (V1.1).smc
[ERROR] [CONTENT LOAD]: Could not read content file "krusty.zip#Krusty's Super Fun House (U) (V1.1).smc"

[INFO] [Core]: Content ran for a total of: 00 hours, 00 minutes, 00 seconds.
[INFO] [Core]: Unloading core..
[INFO] [Core]: Unloading core symbols..
[INFO] [Core Options]: Saved core options file to "~/.config/retroarch/config/Snes9x/Snes9x.opt"
[INFO] [Video]: Does not have enough samples for monitor refresh rate estimation. Requires to run for at least 4096 frames.
```

Extracting the game manually and running the extracted game content with the same command works as expected.
I tested with snes9x and nestopia cores, both are affected. I suspect all other cores are affected as well.

### Expected behavior

I expect the zipped game to launch.

### Actual behavior

The zipped game does not launch.

### Steps to reproduce the bug

1. run a game with: retroarch --libretro /usr/lib/libretro/snes9x_libretro.so game.zip

### Bisect Results

Works with:
retroarch 1.9.3

Does not work with:
retroarch 1.9.4
retroarch 1.9.5

### Version/Commit
You can find this information under Information/System Information

- RetroArch: 1.9.5

### Environment information

- OS: Arch Linux

--- a/libretro-common/file/file_path.c
+++ b/libretro-common/file/file_path.c
@@ -87,8 +87,14 @@ const char *path_get_archive_delim(const char *path)
 
    buf[0] = '\0';
 
+   /* We search for delimiters after the last slash
+    * in the file path to avoid capturing delimiter
+    * characters in any parent directory names.
+    * If there are no slashes in the file name, then
+    * the path is just the file basename - in this
+    * case we search the path in its entirety */
    if (!last_slash)
-      return NULL;
+      last_slash = path;
 
    /* Find delimiter position
     * > Since filenames may contain '#' characters,

